pkg/apis/triggers/v1alpha1/trigger_binding_validation.go

```diff
@@ -18,29 +18,71 @@ package v1alpha1
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	"k8s.io/apimachinery/pkg/util/sets"
 	"knative.dev/pkg/apis"
 )
 
 // Validate TriggerBinding.
 func (tb *TriggerBinding) Validate(ctx context.Context) *apis.FieldError {
-	return tb.Spec.Validate(ctx)
+	return tb.Spec.Validate(ctx).ViaField("spec")
 }
 
 // Validate TriggerBindingSpec.
 func (s *TriggerBindingSpec) Validate(ctx context.Context) *apis.FieldError {
-	return validateParams(s.Params)
+	return validateParams(s.Params).ViaField("params")
 }
 
 func validateParams(params []Param) *apis.FieldError {
 	// Ensure there aren't multiple params with the same name.
 	seen := sets.NewString()
-	for _, param := range params {
+	for i, param := range params {
 		if seen.Has(param.Name) {
-			return apis.ErrMultipleOneOf("spec.params")
+			return apis.ErrMultipleOneOf(fmt.Sprintf("[%d].name", i))
 		}
 		seen.Insert(param.Name)
+		errs := validateParamValue(param.Value).ViaField(fmt.Sprintf("[%d]", i))
+		if errs != nil {
+			return errs
+		}
 	}
 	return nil
 }
+
+func validateParamValue(in string) *apis.FieldError {
+	if !strings.Contains(in, "$(") {
+		return nil
+	}
+	// Splits string on $( to find potential Tekton expressions
+	maybeExpressions := strings.Split(in, "$(")
+	terminated := true
+	for _, e := range maybeExpressions[1:] { // Split always returns at least one element
+		// Iterate until we find the first unbalanced )
+		numOpenBrackets := 0
```

Comment:
question: is it possible to reuse this bit from the jsonpath.go? Maybe this bit could be extracted to its own function? If that's too much for this PR, I'm happy to merge this as is.